cartograph/cluster_builder.py

```diff
@@ -21,34 +29,80 @@ def get_kmeans(article_vectors, k):
     :param k: desired number of clusters for kmeans
     :return: a dataframe two columns: article id and country
     """
-
     article_id = article_vectors['article_id']
-    matrix = article_vectors.iloc[:, 2:].values
+    # creates a 4004 x 100 matrix
+    matrix = article_vectors.iloc[:, 1:].values
     kmeans = KMeans(n_clusters=int(k)).fit(matrix[0:len(matrix)])
     article_vectors['country'] = kmeans.labels_
-
+    # creates a 4004 x 2 dataframe with article id and the country it belongs to.
     df = pd.DataFrame(article_vectors, columns=['country'])
     df['article_id'] = article_id
-    return df.set_index(df['article_id']).iloc[:, 0:1]
-
-
-def get_hdbscan(article_vectors, min_size):
+    centroids = pd.DataFrame(list(kmeans.cluster_centers_))
+    centroids.index.name = 'country'
+    df_with_vectors = pd.merge(df, article_vectors.drop(columns='country'), on='article_id')
+    df_with_centroids = pd.merge(df_with_vectors, centroids, on='country')
+    dist = []
+    for index, row in df_with_centroids.iterrows():
+        a = np.array(row[2:102])
```

Comment:
it's better to not hard code the 100 here. calculate num_dims using the number of columns, or pass it in to the function.